docs/src/main/tut/datatypes/validated.md

```diff
@@ -12,7 +12,351 @@ Response comes back saying your username can't have dashes in it, so you make so
+
+Note that, at the end, we expect to lift the result of the validation functions in a `RegistrationData` instance. If the process fails, we'll get our `NonEmptyList` detailing what went wrong.
+
+For example:
+
+```tut:book
+FormValidatorNel.validateForm(
+  username = "Joe",
+  password = "Passw0r$1234",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
+
+FormValidatorNel.validateForm(
+  username = "Joe%%%",
+  password = "password",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
+```
+
+Sweet success! Now you can take your validation process to the next level!
+
+### A short detour
+
+Typically, you'll see that `Validated` will be accompanied by a `NonEmptyList` when it comes to accumulation. The thing here is that you can define your own accumulative data structure and you're not limited to the aforementioned construction.
+
+For doing this, you have to provide a `Semigroup` instance. `NonEmptyList`, by definition has its own `Semigroup`. For those who don't know what a `Semigroup` is, let's see a simple example.
+
+#### Accumulative Structures
+
+According to [Wikipedia](https://en.wikipedia.org/wiki/Semigroup):
+
+> A semigroup is an algebraic structure consisting of a set together with an associative binary operation.
+
+You can find more about how `Semigroup` works in cats [here](../typeclasses/semigroup.html).
+
+Let's take a look at `ap` method of `Validated`:
+
+```tut:silent:fail
+/**
+   * From Apply:
+   * if both the function and this value are Valid, apply the function
+   */
+def ap[EE >: E, B](f: Validated[EE, A => B])(implicit EE: Semigroup[EE]): Validated[EE, B] =
+  (this, f) match {
+    // ...
+    case (Invalid(e1), Invalid(e2)) => Invalid(EE.combine(e2, e1))
+    // ...
+  }
+}
+```
+
+We've omitted the complete implementation because our focus here is the case in where you need to append (that's the function of this method) two failures. Note the `implicit EE: Semigroup[EE]` parameter and the usage of its `.combine` operation. In the case of `NonEmptyList`, we're talking about a `List`, with certain properties that allow us to _combine_ (append) more than one element to it. That's because, apart from the fact that it is a `List`, it also has an instance of `Semigroup`, telling it how to operate with the accumulation.
+
+As we've said before: if you need another data type for processing the failures, you can use it, providing an instance of a `Semigroup` with the `.combine` logic.
+
+### Going back and forth
+
+cats offer you a nice set of combinators to transform your `Validated` based approach to an `Either` one and vice-versa.
+Please note that, if you're using an `Either`-based approach as seen in our first example and you choose to convert it to a `Validated` one, you're constrained to the fail-fast nature of `Either`, but you're gaining a broader set of features with `Validated`.
```

Comment:
What do you mean with 
> you're gaining a broader set of features with `Validated`.

Reply:
This is the section in where I have doubts about it. I was trying to express that, coming from `Either` to `Validated`, the person will gain more _combinators_ (_Features_), even if you transfer the `Either` behavior (fail-fast) to `Validated`. 

With your previous comments, I have an idea for simplifying this section (or even delete it). Let me work on it and I'll reach you out again :)